diff -crB systemd-232-orig/units/user@.service.m4.in systemd-232/units/user@.service.m4.in
*** systemd-232-orig/units/user@.service.m4.in	2016-11-04 02:16:42.000000000 +0900
--- systemd-232/units/user@.service.m4.in	2016-11-22 15:09:53.764506091 +0900
***************
*** 14,19 ****
--- 14,21 ----
  PAMName=systemd-user
  Type=notify
  ExecStart=-@rootlibexecdir@/systemd --user
+ ExecStop=/usr/bin/kill -TERM ${MAINPID}
+ KillSignal=SIGCONT
  Slice=user-%i.slice
  KillMode=mixed
  Delegate=yes
