diff -crB VirtualBox-4.3.12-orig/src/VBox/Devices/testcase/Makefile.kmk VirtualBox-4.3.12/src/VBox/Devices/testcase/Makefile.kmk
*** VirtualBox-4.3.12-orig/src/VBox/Devices/testcase/Makefile.kmk	Fri May 30 10:59:42 2014
--- VirtualBox-4.3.12/src/VBox/Devices/testcase/Makefile.kmk	Fri May 30 15:55:54 2014
***************
*** 66,71 ****
--- 66,72 ----
   #
   PROGRAMS += tstDeviceStructSizeRC
   tstDeviceStructSizeRC_TEMPLATE  = VBoxRcExe
+  tstDeviceStructSizeRC_LDFLAGS   = -lrt
   tstDeviceStructSizeRC_DEFS      = $(VBOX_DEVICES_TESTS_FEATURES)
   tstDeviceStructSizeRC_SOURCES   = tstDeviceStructSizeRC.cpp
   tstDeviceStructSizeRC_INCS      = \
diff -crB VirtualBox-4.3.12-orig/src/VBox/VMM/testcase/Makefile.kmk VirtualBox-4.3.12/src/VBox/VMM/testcase/Makefile.kmk
*** VirtualBox-4.3.12-orig/src/VBox/VMM/testcase/Makefile.kmk	Fri May 30 10:59:41 2014
--- VirtualBox-4.3.12/src/VBox/VMM/testcase/Makefile.kmk	Fri May 30 16:39:38 2014
***************
*** 123,128 ****
--- 123,129 ----
  #
  ifdef VBOX_WITH_RAW_MODE
   tstVMStructRC_TEMPLATE  = VBoxRcExe
+  tstVMStructRC_LDFLAGS   = -lrt
   tstVMStructRC_DEFS      = IN_VMM_RC IN_DIS IN_RT_RC VBOX_WITH_RAW_MODE
   ifeq ($(KBUILD_TARGET).$(KBUILD_TARGET_ARCH),darwin.x86)
    tstVMStructRC_DEFS    += \
***************
*** 185,190 ****
--- 186,192 ----
  
  ifdef VBOX_WITH_RAW_MODE
   tstAsmStructsRC_TEMPLATE= VBoxRcExe
+  tstAsmStructsRC_LDFLAGS = -lrt
   tstAsmStructsRC_DEFS    = IN_VMM_RC IN_DIS IN_RT_RC VBOX_WITH_RAW_MODE
   ifdef VBOX_WITH_R0_LOGGING
    tstAsmStructsRC_DEFS  += VBOX_WITH_R0_LOGGING
