diff -crB NetworkManager-0.9.7.997-orig/policy/org.freedesktop.NetworkManager.policy.in.in NetworkManager-0.9.7.997/policy/org.freedesktop.NetworkManager.policy.in.in
*** NetworkManager-0.9.7.997-orig/policy/org.freedesktop.NetworkManager.policy.in.in	Wed Jan 23 18:37:11 2013
--- NetworkManager-0.9.7.997/policy/org.freedesktop.NetworkManager.policy.in.in	Fri Feb 15 21:39:15 2013
***************
*** 13,19 ****
      <_description>Enable or disable system networking</_description>
      <_message>System policy prevents enabling or disabling system networking</_message>
      <defaults>
!       <allow_inactive>no</allow_inactive>
        <allow_active>yes</allow_active>
      </defaults>
    </action>
--- 13,19 ----
      <_description>Enable or disable system networking</_description>
      <_message>System policy prevents enabling or disabling system networking</_message>
      <defaults>
!       <allow_inactive>yes</allow_inactive>
        <allow_active>yes</allow_active>
      </defaults>
    </action>
***************
*** 22,29 ****
      <_description>Put NetworkManager to sleep or wake it up (should only be used by system power management)</_description>
      <_message>System policy prevents putting NetworkManager to sleep or waking it up</_message>
      <defaults>
!       <allow_inactive>no</allow_inactive>
!       <allow_active>no</allow_active>
      </defaults>
    </action>
  
--- 22,29 ----
      <_description>Put NetworkManager to sleep or wake it up (should only be used by system power management)</_description>
      <_message>System policy prevents putting NetworkManager to sleep or waking it up</_message>
      <defaults>
!       <allow_inactive>yes</allow_inactive>
!       <allow_active>yes</allow_active>
      </defaults>
    </action>
  
***************
*** 31,37 ****
      <_description>Enable or disable WiFi devices</_description>
      <_message>System policy prevents enabling or disabling WiFi devices</_message>
      <defaults>
!       <allow_inactive>no</allow_inactive>
        <allow_active>yes</allow_active>
      </defaults>
    </action>
--- 31,37 ----
      <_description>Enable or disable WiFi devices</_description>
      <_message>System policy prevents enabling or disabling WiFi devices</_message>
      <defaults>
!       <allow_inactive>yes</allow_inactive>
        <allow_active>yes</allow_active>
      </defaults>
    </action>
***************
*** 40,46 ****
      <_description>Enable or disable mobile broadband devices</_description>
      <_message>System policy prevents enabling or disabling mobile broadband devices</_message>
      <defaults>
!       <allow_inactive>no</allow_inactive>
        <allow_active>yes</allow_active>
      </defaults>
    </action>
--- 40,46 ----
      <_description>Enable or disable mobile broadband devices</_description>
      <_message>System policy prevents enabling or disabling mobile broadband devices</_message>
      <defaults>
!       <allow_inactive>yes</allow_inactive>
        <allow_active>yes</allow_active>
      </defaults>
    </action>
***************
*** 49,55 ****
      <_description>Enable or disable WiMAX mobile broadband devices</_description>
      <_message>System policy prevents enabling or disabling WiMAX mobile broadband devices</_message>
      <defaults>
!       <allow_inactive>no</allow_inactive>
        <allow_active>yes</allow_active>
      </defaults>
    </action>
--- 49,55 ----
      <_description>Enable or disable WiMAX mobile broadband devices</_description>
      <_message>System policy prevents enabling or disabling WiMAX mobile broadband devices</_message>
      <defaults>
!       <allow_inactive>yes</allow_inactive>
        <allow_active>yes</allow_active>
      </defaults>
    </action>
***************
*** 67,73 ****
      <_description>Connection sharing via a protected WiFi network</_description>
      <_message>System policy prevents sharing connections via a protected WiFi network</_message>
      <defaults>
!       <allow_inactive>no</allow_inactive>
        <allow_active>yes</allow_active>
      </defaults>
    </action>
--- 67,73 ----
      <_description>Connection sharing via a protected WiFi network</_description>
      <_message>System policy prevents sharing connections via a protected WiFi network</_message>
      <defaults>
!       <allow_inactive>yes</allow_inactive>
        <allow_active>yes</allow_active>
      </defaults>
    </action>
***************
*** 76,82 ****
      <_description>Connection sharing via an open WiFi network</_description>
      <_message>System policy prevents sharing connections via an open WiFi network</_message>
      <defaults>
!       <allow_inactive>no</allow_inactive>
        <allow_active>yes</allow_active>
      </defaults>
    </action>
--- 76,82 ----
      <_description>Connection sharing via an open WiFi network</_description>
      <_message>System policy prevents sharing connections via an open WiFi network</_message>
      <defaults>
!       <allow_inactive>yes</allow_inactive>
        <allow_active>yes</allow_active>
      </defaults>
    </action>
***************
*** 85,91 ****
      <_description>Modify personal network connections</_description>
      <_message>System policy prevents modification of personal network settings</_message>
      <defaults>
!       <allow_inactive>no</allow_inactive>
        <allow_active>yes</allow_active>
      </defaults>
    </action>
--- 85,91 ----
      <_description>Modify personal network connections</_description>
      <_message>System policy prevents modification of personal network settings</_message>
      <defaults>
!       <allow_inactive>yes</allow_inactive>
        <allow_active>yes</allow_active>
      </defaults>
    </action>
***************
*** 94,101 ****
      <_description>Modify network connections for all users</_description>
      <_message>System policy prevents modification of network settings for all users</_message>
      <defaults>
!       <allow_inactive>no</allow_inactive>
!       <allow_active>@NM_MODIFY_SYSTEM_POLICY@</allow_active>
      </defaults>
    </action>
  
--- 94,101 ----
      <_description>Modify network connections for all users</_description>
      <_message>System policy prevents modification of network settings for all users</_message>
      <defaults>
!       <allow_inactive>yes</allow_inactive>
!       <allow_active>yes</allow_active>
      </defaults>
    </action>
  
***************
*** 103,110 ****
      <_description>Modify persistent system hostname</_description>
      <_message>System policy prevents modification of the persistent system hostname</_message>
      <defaults>
!       <allow_inactive>no</allow_inactive>
!       <allow_active>auth_admin_keep</allow_active>
      </defaults>
    </action>
  
--- 103,110 ----
      <_description>Modify persistent system hostname</_description>
      <_message>System policy prevents modification of the persistent system hostname</_message>
      <defaults>
!       <allow_inactive>yes</allow_inactive>
!       <allow_active>yes</allow_active>
      </defaults>
    </action>
  
diff -crB NetworkManager-0.9.7.997-orig/src/org.freedesktop.NetworkManager.conf NetworkManager-0.9.7.997/src/org.freedesktop.NetworkManager.conf
*** NetworkManager-0.9.7.997-orig/src/org.freedesktop.NetworkManager.conf	Sun Nov  4 22:28:53 2012
--- NetworkManager-0.9.7.997/src/org.freedesktop.NetworkManager.conf	Fri Feb 15 21:40:03 2013
***************
*** 69,94 ****
                  <allow send_destination="org.freedesktop.NetworkManager"
                         send_interface="org.freedesktop.NetworkManager.AgentManager"/>
  
!                 <deny send_destination="org.freedesktop.NetworkManager"
                         send_interface="org.freedesktop.NetworkManager"
                         send_member="SetLogging"/>
  
!                 <deny send_destination="org.freedesktop.NetworkManager"
                         send_interface="org.freedesktop.NetworkManager"
                         send_member="Sleep"/>
  
!                 <deny send_destination="org.freedesktop.NetworkManager"
                         send_interface="org.freedesktop.NetworkManager"
                         send_member="sleep"/>
  
!                 <deny send_destination="org.freedesktop.NetworkManager"
                         send_interface="org.freedesktop.NetworkManager"
                         send_member="wake"/>
          </policy>
          <policy context="default">
!                 <deny own="org.freedesktop.NetworkManager"/>
  
!                 <deny send_destination="org.freedesktop.NetworkManager"/>
  
                  <allow send_destination="org.freedesktop.NetworkManager"
                         send_interface="org.freedesktop.NetworkManager.Settings"/>
--- 69,94 ----
                  <allow send_destination="org.freedesktop.NetworkManager"
                         send_interface="org.freedesktop.NetworkManager.AgentManager"/>
  
!                 <allow send_destination="org.freedesktop.NetworkManager"
                         send_interface="org.freedesktop.NetworkManager"
                         send_member="SetLogging"/>
  
!                 <allow send_destination="org.freedesktop.NetworkManager"
                         send_interface="org.freedesktop.NetworkManager"
                         send_member="Sleep"/>
  
!                 <allow send_destination="org.freedesktop.NetworkManager"
                         send_interface="org.freedesktop.NetworkManager"
                         send_member="sleep"/>
  
!                 <allow send_destination="org.freedesktop.NetworkManager"
                         send_interface="org.freedesktop.NetworkManager"
                         send_member="wake"/>
          </policy>
          <policy context="default">
!                 <allow own="org.freedesktop.NetworkManager"/>
  
!                 <allow send_destination="org.freedesktop.NetworkManager"/>
  
                  <allow send_destination="org.freedesktop.NetworkManager"
                         send_interface="org.freedesktop.NetworkManager.Settings"/>
***************
*** 96,114 ****
                  <allow send_destination="org.freedesktop.NetworkManager"
                         send_interface="org.freedesktop.NetworkManager.AgentManager"/>
  
!                 <deny send_destination="org.freedesktop.NetworkManager"
                         send_interface="org.freedesktop.NetworkManager"
                         send_member="SetLogging"/>
  
!                 <deny send_destination="org.freedesktop.NetworkManager"
                         send_interface="org.freedesktop.NetworkManager"
                         send_member="Sleep"/>
  
!                 <deny send_destination="org.freedesktop.NetworkManager"
                         send_interface="org.freedesktop.NetworkManager"
                         send_member="sleep"/>
  
!                 <deny send_destination="org.freedesktop.NetworkManager"
                         send_interface="org.freedesktop.NetworkManager"
                         send_member="wake"/>
          </policy>
--- 96,114 ----
                  <allow send_destination="org.freedesktop.NetworkManager"
                         send_interface="org.freedesktop.NetworkManager.AgentManager"/>
  
!                 <allow send_destination="org.freedesktop.NetworkManager"
                         send_interface="org.freedesktop.NetworkManager"
                         send_member="SetLogging"/>
  
!                 <allow send_destination="org.freedesktop.NetworkManager"
                         send_interface="org.freedesktop.NetworkManager"
                         send_member="Sleep"/>
  
!                 <allow send_destination="org.freedesktop.NetworkManager"
                         send_interface="org.freedesktop.NetworkManager"
                         send_member="sleep"/>
  
!                 <allow send_destination="org.freedesktop.NetworkManager"
                         send_interface="org.freedesktop.NetworkManager"
                         send_member="wake"/>
          </policy>
