diff -crB cyrus-sasl-2.1.26-orig/include/sasl.h cyrus-sasl-2.1.26/include/sasl.h
*** cyrus-sasl-2.1.26-orig/include/sasl.h	Fri Oct 12 14:05:48 2012
--- cyrus-sasl-2.1.26/include/sasl.h	Thu Feb  7 20:07:42 2013
***************
*** 130,135 ****
--- 130,136 ----
  #define SASL_VERSION_FULL ((SASL_VERSION_MAJOR << 16) |\
        (SASL_VERSION_MINOR << 8) | SASL_VERSION_STEP)
  
+ #include <sys/types.h>
  #include "prop.h"
  
  /*************
